### PR TITLE
Add Tizen 8.0 bindings

### PR DESCRIPTION
**Pull request by JSUYA**
The rootstrap path in tizen 8.0 has changed.

API reference: https://docs.tizen.org/application/native/api/common/8.0/
[Updated headers in 8.0]
nntrainer/nntrainer.h
nnstreamer/ml-api-service.h
media/mv_common.h
media/webrtc.h
media/sound_manager.h
media/player.h
media/camera.h
tts.h
rpc-port/rpc-port-parcel.h
stt.h
appfw/app_common.h
appfw/app_control.h
appfw/app.h
web/download.h
network/net_connection.h
network/bluetooth.h
network/wifi-manager.h
peripheral_io.h

### CODE REVIEW

File: configs/8.0/ffigen.yaml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    - '**/vce.h'
+    - '**/voice_control.h'
+    - '**/voice_control_manager.h'
+    - '**/vpn_service.h'
+    - '**/appcore-agent/*.h'
+    - '**/appfw/*.h'
+    - '**/asp/*.h'
+    - '**/badge/*.h'
+    - '**/calendar-service2/*.h'
+    - '**/cbhm/*.h'
```
Comment (swift-kim) [inline configs/8.0/ffigen.yaml:57]: Somehow cbhm is still present?

File: lib/src/bindings/5.0/generated_bindings.dart [hunk 2]
```diff
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-// ignore_for_file: type=lint, unused_element, unused_field
+// ignore_for_file: type=lint, unused_element, unused_field, subtype_of_base_or_final_is_not_base_final_or_sealed
```
Comment (swift-kim) [inline lib/src/bindings/5.0/generated_bindings.dart:5]: Shouldn't the original `ffigen.yaml` (for Tizen 7.0 and older) be updated if this is necessary?

File: lib/src/bindings/4.0/generated_bindings.dart [hunk 3]
```diff
@@ -2759,9 +2759,9 @@ class Tizen40Native {
   }
 
   late final _notification_set_time_to_textPtr = _lookup<
-      ffi.NativeFunction<
-          ffi.Int Function(notification_h, ffi.Int32,
-              time_t)>>('notification_set_time_to_text');
+          ffi
+          .NativeFunction<ffi.Int Function(notification_h, ffi.Int32, time_t)>>(
+      'notification_set_time_to_text');
```
Comment (swift-kim) [inline lib/src/bindings/4.0/generated_bindings.dart:2764]: Are these only formatting changes?

File: scripts/copy_rootstrap.sh [hunk 4]
```diff
@@ -23,9 +23,19 @@ if [[ -z $TIZEN_SDK ]]; then
   fi
 fi
 
-rootstrap=$TIZEN_SDK/platforms/tizen-$version/iot-headed/rootstraps/iot-headed-$version-device.core
+if [ "$version" == "8.0" ]
+then
```
Comment (swift-kim) [inline scripts/copy_rootstrap.sh:27]: For consistency and portability (`==` is not posix standard):
```suggestion
if [ "$version" = "8.0" ]; then
```

File: scripts/copy_rootstrap.sh [hunk 5]
```diff
@@ -23,9 +23,19 @@ if [[ -z $TIZEN_SDK ]]; then
...
+  rootstrap=$TIZEN_SDK/platforms/tizen-$version/tizen/rootstraps/tizen-$version-device.core
+else
+  rootstrap=$TIZEN_SDK/platforms/tizen-$version/iot-headed/rootstraps/iot-headed-$version-device.core
+fi
 if [ ! -d $rootstrap ]; then
-  echo "Rootstrap not installed: IOT-Headed-$version-NativeAppDevelopment-CLI"
+  if [ "$version" == "8.0"]
```
Comment (swift-kim) [inline scripts/copy_rootstrap.sh:33]: nit: Missing space
```suggestion
  if [ "$version" = "8.0" ]
```

File: README.md [hunk 6]
```diff
@@ -80,147 +80,147 @@ tizen.storage_foreach_device_supported(
 
 ## Supported APIs
 
-This package provides bindings for the following APIs of the [Tizen IoT-Headed](https://docs.tizen.org/application/native/api/iot-headed/latest) profile.
+This package provides bindings for the following APIs of the [Tizen IoT-Headed](https://docs.tizen.org/application/native/api/iot-headed/latest) profile ([In 8.0, Tizen Common](https://docs.tizen.org/application/native/api/common/8.0/index.html) profile).
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply README.md]: done

File: README.md [hunk 7]
```diff
@@ -80,147 +80,147 @@ tizen.storage_foreach_device_supported(
...
-| | App Control URI | | | | | ✔ | ✔ |
-| | Event | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |
-| | Internationalization | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |
-| | Job scheduler | | ✔ | ✔ | ✔ | ✔ | ✔ |
-| | Preference | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |
-| | Resource Manager | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |
-| | Application Manager | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |
+| Category | Sub category | Tizen 4.0 | Tizen 5.0 | Tizen 5.5 | Tizen 6.0 | Tizen 6.5 | Tizen 7.0 | Tizen 8.0 |
+|-|-|:-:|:-:|:-:|:-:|:-:|:-:|:-:|
+| Account | Account Manager | | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ | ✔ |
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply README.md]: my mistake. I fixed it

File: configs/8.0/entrypoints.h [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+#include <mime_type.h>
+#include <media_content.h>
+
+// Context
+#include <context_history.h>
+#include <context_trigger.h>
+
+// Location
+#include <geofence_manager.h> // will deprecated
+#include <geofence_type.h> // will deprecated
```
[the remark this answers is not in the document]
Comment (JSUYA) [reply configs/8.0/entrypoints.h]: done

File: configs/8.0/ffigen.yaml (hunk 1 above)
Comment (JSUYA) [reply configs/8.0/ffigen.yaml]: removed

File: lib/src/bindings/5.0/generated_bindings.dart (hunk 2 above)
Comment (JSUYA) [reply lib/src/bindings/5.0/generated_bindings.dart]: I added ffigen.yaml files

File: lib/src/bindings/4.0/generated_bindings.dart (hunk 3 above)
Comment (JSUYA) [reply lib/src/bindings/4.0/generated_bindings.dart:2764]: The format has changed. In my view, there doesn't seem to be any API change.

File: scripts/copy_rootstrap.sh (hunk 4 above)
Comment (JSUYA) [reply scripts/copy_rootstrap.sh]: done

File: scripts/copy_rootstrap.sh (hunk 5 above)
Comment (JSUYA) [reply scripts/copy_rootstrap.sh]: done

File: lib/src/bindings/4.0/generated_bindings.dart (hunk 3 above)
Comment (swift-kim) [reply lib/src/bindings/4.0/generated_bindings.dart:2764]: We didn't update the ffigen version. What made the changes?

→ ffigen 7.2.10 added parameter names for function pointer arguments.

> 7.2.10 [#](https://pub.dev/packages/ffigen/changelog#7210) 
> Generate parameter names in function pointer fields and typedefs.

You can update the dev_dependency in `pubspec.yaml` to the version installed on your machine if you want to be clear.

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (swift-kim) [reply README.md]: Please review again. You made another serious mistake :)

[the remark this answers is not in the document]
Comment (JSUYA) [reply README.md]: sorry again. i updated it

File: lib/src/bindings/4.0/generated_bindings.dart (hunk 3 above)
Comment (JSUYA) [reply lib/src/bindings/4.0/generated_bindings.dart:2764]: Thank you for your review. 
I updated ffigen version to latest(11.0.0). And I reverted the ffigen.yaml(4.0 ~ 7.0) changes.
 +) I updated the version of the tizen_interop_callbacks package to 0.1.1.
please let me know if there's more to add.

Comment (swift-kim) [reply lib/src/bindings/4.0/generated_bindings.dart:2764]: Tip: Please check out the changelog and README of the ffigen package for newly added, potentially useful configuration options and other features when updating the ffigen version.

**Review by swift-kim — APPROVED** [review]

File: lib/src/bindings/4.0/generated_bindings.dart (hunk 3 above)
Comment (JSUYA) [reply lib/src/bindings/4.0/generated_bindings.dart:2764]: As far as I know, the minimum version we need from ffigen is 8.0.0-dev.0.
( 8.0.0-dev.0 : Adds final class modifier to generated sub types Struct, Union and Opaque...)
If we use ffigen 8.0.0, we need to adjust `file` package version, but `file` package applies the fixes made from dart 3.0 in 7.0.0.
Therefore, in order to satisfy the version, ffigen version should be 11.0.0.

**Pull request merged**